### PR TITLE
Decrement withdrawalRequestAmount after withdrawal passes, replace `onlyCollateralizedPosition` with `_getPositionData`, address some TODO's

### PR DESCRIPTION
**Pull request by nicholaspai**
- `_getPositionData(sponsor)` already has the modifier `onlyCollateralizedPosition(sponsor)`, therefore there is no need to check for "valid" positions (i.e. positions with collateral) outside of calling `_getPositionData(sponsor)`
- Decrement `withdrawalRequestAmount` to 0 after calling `withdrawPassedRequest`

Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol [hunk 1]
```diff
@@ -39,7 +39,6 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         FixedPoint.Unsigned rawCollateral;
     }
 
-    // TODO: determine how we should adjust collateral when the user reads it out of the contract.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: addressed by exposing `getCollateralPosition` which returns collateral net of fees

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol [hunk 2]
```diff
@@ -154,7 +153,6 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
      * @param collateralAmount represents the total amount of tokens to be sent to the position for the sponsor.
      */
 
-    // TODO: should this check if the position is valid first?
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: already handled by `_getPositionData`

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol [hunk 3]
```diff
@@ -222,8 +220,6 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
      * @notice After a passed withdrawal request (i.e., by a call to `requestWithdrawal` and waiting
      * `withdrawalLiveness`), withdraws `positionData.withdrawalRequestAmount` of collateral currency.
      */
-    // TODO: is onlyCollateralizedPosition(msg.sender) correct here? if a position withdraws all their collateral will this still work?
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: @chrismaree how would you replicate this in a test?

Comment (nicholaspai) [reply core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: FWIW: I believe that `onlyCollateralizedPosition` does work as intended: checking if collateral > 0

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol [hunk 4]
```diff
@@ -328,8 +326,6 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
      * @notice After expiration of the contract the DVM is asked what for the prevailing price at the time of
      * expiration. In addition, pay the final fee at this time. Once this has been resolved token holders can withdraw.
      */
-    // TODO: what are the implications of calling this multiple times? can this not drain the contract by paying
-    // the final fee over and over again?
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: Addressed by setting `ContractState`

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol [hunk 5]
```diff
@@ -222,22 +211,22 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
...
-        positionData.requestPassTimestamp = 0;
-
         // Transfer approved withdrawal amount from the contract to the caller.
         collateralCurrency.safeTransfer(msg.sender, positionData.withdrawalRequestAmount.rawValue);
 
         emit RequestWithdrawalExecuted(msg.sender, positionData.withdrawalRequestAmount.rawValue);
+
+        // Reset withdrawal request
+        positionData.withdrawalRequestAmount = FixedPoint.fromUnscaledUint(0);
+        positionData.requestPassTimestamp = 0;
```
Comment (chrismaree) [inline core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: This is a good catch. I actually noticed this problem when working through the truffle migration process with Regina.

File: core/contracts/financial-templates/implementation/PricelessPositionManager.sol (hunk 3 above)
Comment (chrismaree) [reply core/contracts/financial-templates/implementation/PricelessPositionManager.sol]: ye, I agree

**Comment by nicholaspai** [comment]
> This looks good! however do we have enough test coverage here to check that the values are set to zero after the withdrawal passes? It clearly is set but the test does not check it

Great catch, I've added these tests to `PricelessPositionManager.js` "Withdrawal request" test

**Pull request merged**